packages/kokkos/core/unit_test/TestDeepCopy.hpp

```diff
@@ -0,0 +1,167 @@
+    {
+      int a_begin = 3;
+      int a_end   = 0;
+      int b_begin = 3;
+      int b_end   = 0;
+      auto a = Kokkos::subview(a_char,std::pair<int,int>(a_begin,a_char.extent(0)-a_end));
+      auto b = Kokkos::subview(b_char,std::pair<int,int>(b_begin,b_char.extent(0)-b_end));
+      auto a_copy = Kokkos::subview(a_char_copy,std::pair<int,int>(a_begin,a_char_copy.extent(0)-a_end));
+      Kokkos::deep_copy(b,a);
+      Kokkos::deep_copy(a_copy,b);
+      int check = compare_equal(a_copy,a);
+      ASSERT_EQ( check, 0 );
+    }
+
+    {
+      int a_begin = 3;
+      int a_end   = 6;
+      int b_begin = 3;
+      int b_end   = 6;
+      auto a = Kokkos::subview(a_char,std::pair<int,int>(a_begin,a_char.extent(0)-a_end));
+      auto b = Kokkos::subview(b_char,std::pair<int,int>(b_begin,b_char.extent(0)-b_end));
+      auto a_copy = Kokkos::subview(a_char_copy,std::pair<int,int>(a_begin,a_char_copy.extent(0)-a_end));
+      Kokkos::deep_copy(b,a);
+      Kokkos::deep_copy(a_copy,b);
+      int check = compare_equal(a_copy,a);
+      ASSERT_EQ( check, 0 );
+    }
+
+    {
+      int a_begin = 5;
+      int a_end   = 4;
+      int b_begin = 3;
+      int b_end   = 6;
+      auto a = Kokkos::subview(a_char,std::pair<int,int>(a_begin,a_char.extent(0)-a_end));
+      auto b = Kokkos::subview(b_char,std::pair<int,int>(b_begin,b_char.extent(0)-b_end));
+      auto a_copy = Kokkos::subview(a_char_copy,std::pair<int,int>(a_begin,a_char_copy.extent(0)-a_end));
+      Kokkos::deep_copy(b,a);
+      Kokkos::deep_copy(a_copy,b);
+      int check = compare_equal(a_copy,a);
+      ASSERT_EQ( check, 0 );
+    }
+
+    {
+      int a_begin = 0;
+      int a_end   = 8;
+      int b_begin = 2;
+      int b_end   = 6;
+      auto a = Kokkos::subview(a_char,std::pair<int,int>(a_begin,a_char.extent(0)-a_end));
+      auto b = Kokkos::subview(b_char,std::pair<int,int>(b_begin,b_char.extent(0)-b_end));
+      auto a_copy = Kokkos::subview(a_char_copy,std::pair<int,int>(a_begin,a_char_copy.extent(0)-a_end));
+      Kokkos::deep_copy(b,a);
+      Kokkos::deep_copy(a_copy,b);
+      int check = compare_equal(a_copy,a);
+      ASSERT_EQ( check, 0 );
+    }
+     
+    {
+      int a_begin = 2;
+      int a_end   = 6;
+      int b_begin = 0;
+      int b_end   = 8;
+      auto a = Kokkos::subview(a_char,std::pair<int,int>(a_begin,a_char.extent(0)-a_end));
+      auto b = Kokkos::subview(b_char,std::pair<int,int>(b_begin,b_char.extent(0)-b_end));
+      auto a_copy = Kokkos::subview(a_char_copy,std::pair<int,int>(a_begin,a_char_copy.extent(0)-a_end));
+      Kokkos::deep_copy(b,a);
+      Kokkos::deep_copy(a_copy,b);
+      int check = compare_equal(a_copy,a);
+      ASSERT_EQ( check, 0 );
+    }
+
+  }
+};
+}
+
+TEST_F( TEST_CATEGORY, deep_copy_alignment )
+{
+  { Impl::TestDeepCopy< TEST_EXECSPACE::memory_space , TEST_EXECSPACE::memory_space >::run_test( 100000 ); }
+  { Impl::TestDeepCopy< Kokkos::HostSpace , TEST_EXECSPACE::memory_space >::run_test( 100000 ); }
```

Comment:
FYI, we have been changing some of the O(100K) tests to O(1K) tests in Kokkos so debug tests actually pass. This might be ok as it is a copy, still you never know when multiple tests are running together in debug mode.